test/Microsoft.AspNet.Razor.Test/Generator/CSharpTagHelperRenderingTest.cs

```diff
@@ -59,50 +30,64 @@ public static TheoryData TagHelperDescriptorFlowTestData
                     {
                         "SingleTagHelper",
                         "SingleTagHelper",
-                        PAndInputTagHelperDescriptors,
-                        PAndInputTagHelperDescriptors,
+                        DefaultPAndInputTagHelperDescriptors,
+                        DefaultPAndInputTagHelperDescriptors,
                         false
                     },
                     {
                         "SingleTagHelper",
                         "SingleTagHelper.DesignTime",
-                        PAndInputTagHelperDescriptors,
-                        PAndInputTagHelperDescriptors,
+                        DefaultPAndInputTagHelperDescriptors,
+                        DefaultPAndInputTagHelperDescriptors,
                         true
                     },
                     {
                         "BasicTagHelpers",
                         "BasicTagHelpers",
-                        PAndInputTagHelperDescriptors,
-                        PAndInputTagHelperDescriptors,
+                        DefaultPAndInputTagHelperDescriptors,
+                        DefaultPAndInputTagHelperDescriptors,
                         false
                     },
                     {
                         "BasicTagHelpers",
                         "BasicTagHelpers.DesignTime",
-                        PAndInputTagHelperDescriptors,
-                        PAndInputTagHelperDescriptors,
+                        DefaultPAndInputTagHelperDescriptors,
+                        DefaultPAndInputTagHelperDescriptors,
                         true
                     },
                     {
                         "BasicTagHelpers.RemoveTagHelper",
                         "BasicTagHelpers.RemoveTagHelper",
-                        PAndInputTagHelperDescriptors,
+                        DefaultPAndInputTagHelperDescriptors,
                         Enumerable.Empty<TagHelperDescriptor>(),
                         false
                     },
                     {
+                        "BasicTagHelpers.Prefixed",
+                        "BasicTagHelpers.Prefixed",
+                        BuildPAndInputTagHelperDescriptors(prefix: "th:"),
+                        BuildPAndInputTagHelperDescriptors(prefix: "th:"),
```

Comment:
- vary the valid prefixes in these more end-to-end tests as well
- nit: create a property containing the descriptors with this prefix.  (could do that using a deep copy of `DefaultPAndInputTagHelperDescriptors` instead of churning that code.)